src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -769,9 +770,50 @@ HRESULT _stdcall TerminalResize(void* terminal, COORD dimensions)
     publicTerminal->_terminal->ClearSelection();
     publicTerminal->_renderer->TriggerRedrawAll();
 
+    // Resize the text buffer without changing the renderer draw size.
     return publicTerminal->_terminal->UserResize(dimensions);
 }
 
+/// <summary>
+/// Resizes the renderer to fit the provided width and height without changing the text
+/// buffer size.
+/// </summary>
+HRESULT _stdcall ResizeRendererDrawSpace(void* terminal, double width, double height, _Out_ COORD* dimensions)

```

Comment:
You should test and return failure if `void* terminal` is null before it is cast and has a potential null-deref below.